leafletmap/src/main/resources/leafletmap/leafletmap.html

```diff
@@ -25,6 +25,24 @@
     <script src="leaflet/leaflet.js"></script>
 
     <script src="leaflet-color-markers/leaflet-color-markers.js"></script>
+
+    <!-- helper function to remove all polylines from a map
+    https://stackoverflow.com/questions/14585688/clear-all-polylines-from-leaflet-map/14593494#14593494
+    https://stackoverflow.com/questions/30850344/leaflet-check-if-object-is-path-or-marker/30852790#30852790 -->
+    <script>
+        function clearMarkerAndPaths() {
+            for(i in myMap._layers) {
+                if(myMap._layers[i] instanceof L.Marker || myMap._layers[i] instanceof L.Path) {
+                    try {
+                        myMap.removeLayer(myMap._layers[i]);
+                    }
+                    catch(e) {
+                        console.log("problem with " + e + myMap._layers[i]);
+                    }
+                }
+            }
+        }
+    </script>
```

Comment:
All JavaScript methods and commands were in the Kotlin source code up to now, I would not like to mix this up. There should be one places where they are all defined. Would you please move it to the LeafletMapView class? Take a look at the executeMapSetupScripts() method, there you'll see how to handle multiline scripts easily.

Reply:
Fine with me. Will try to make it work in kotline :-)

I can't see an example where you define a function for further use - all examples in executeMapSetupScripts() directly execute commands. I have been adding javascript functions into html from code previously but thats a tad more involved (at least in java):

```
final String scriptCmd =
"var script = document.createElement(\"script\"); " + 
"script.type = \"text/javascript\"; " + 
"script.text = \"" + script +"\";" + 
"document.getElementsByTagName('head')[0].appendChild(script);";

execScript(scriptCmd);
```

To me it looks a lot uglier than having javascript code in html. Especially, since you already have that anyways... <script src="leaflet/leaflet.js"></script> is basically the same.

Reply:
Done but ugly:

- I still don't speak kotlin so its my java code translated 1:1
- If you don't want to have all js in one long line you need to do something similar to javas "StringEscapeUtils.escapeEcmaScript", otherwise you get "unexpected EOF" errors in js

Reply:
I've meant something like this:

execScript("""
    |var myMap = L.map('map', {
    |    center: new L.LatLng(${cfg.initialCenter.latitude}, ${cfg.initialCenter.longitude}),
    |    zoom: 8,
    |    zoomControl: false,
    |    layers: [layer1]
    |});
    |
    |var attribution = myMap.attributionControl;
    |attribution.setPrefix('Leaflet');""".trimMargin())

I know, it's still not perfect, but much more readable than the line by line string concatenation. And you can place variables inline.